### PR TITLE
rfctr: prepare for adding metadata.orig_elements field

### PR DESCRIPTION
**Pull request by scanny**
**Summary**
Some typing modernization in `elements.py` which will get changes to add the `orig_elements` metadata field.

Also some additions to `unit_util.py` to enable simplified mocking that will be required in the next PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,6 +31,7 @@ select = [
 ]
 ignore = [
     "COM812",   # -- over aggressively insists on trailing commas where not desireable --
+    "PT005",    # -- flags mock fixtures with names intentionally matching private method name --
     "PT011",    # -- pytest.raises({exc}) too broad, use match param or more specific exception --
```
Comment (Coniferish) [inline pyproject.toml:35]: Does this change need to be replicated in in `.pre-commit-config.yaml` and `Makefile`? (note at the top)

File: test_unstructured/unit_utils.py [hunk 2]
```diff
@@ -1,13 +1,42 @@
...
+    "Mock",
+    "call",
+    "class_mock",
+    "function_mock",
+    "initializer_mock",
+    "instance_mock",
+    "method_mock",
+    "property_mock",
+)
+
```
Comment (Coniferish) [inline test_unstructured/unit_utils.py:39]: Not that this needs to be changed, but isn't `__all__` used to define what's imported when you use * to import, or is there another use?

File: test_unstructured/unit_utils.py [hunk 3]
```diff
@@ -54,3 +83,136 @@ def example_doc_path(file_name: str) -> str:
 def parse_optional_datetime(datetime_str: Optional[str]) -> Optional[dt.datetime]:
     """Parse `datetime_str` to a datetime.datetime instance or None if `datetime_str` is None."""
     return dt.datetime.fromisoformat(datetime_str) if datetime_str else None
+
+
+# ------------------------------------------------------------------------------------------------
+# MOCKING FIXTURES
+# ------------------------------------------------------------------------------------------------
+# These allow full-featured and type-save mocks to be created simply by adding a unit-test
```
Comment (Coniferish) [inline test_unstructured/unit_utils.py:91]: type-save or type-safe?

File: pyproject.toml (hunk 1 above)
Comment (scanny) [reply pyproject.toml:35]: Hmm, yeah, good call. I've removed those redundant explicit options definitions. Now they should be picked up from the `pyproject.toml` so it's the single source of truth on options.

File: test_unstructured/unit_utils.py (hunk 2 above)
Comment (scanny) [reply test_unstructured/unit_utils.py:39]: Yes, it is, but it also indicates what you are intentionally "exporting" and prevents ruff errors related to "xyz is imported but not used". That's why I have it here.

File: test_unstructured/unit_utils.py (hunk 3 above)
Comment (scanny) [reply test_unstructured/unit_utils.py]: Yep, good catch. Fixed.

**Review by Coniferish — APPROVED** [review]
LGTM

**Pull request merged**